desk/app/activity.hoon

```diff
@@ -838,6 +699,21 @@
   ^+  cor
   =.  allowed  na
   (give-update [%allow-notifications na] ~)
+++  get-parents
+  |=  =source:a
+  ^-  (list source:a)
+  ?:  ?=(%base -.source)  ~
+  ?<  ?=(%base -.source)
+  :-  [%base ~]
+  ?+  -.source  ~
+    %channel  ~[[%group group.source]]
+    %dm-thread  ~[[%dm whom.source]]
+  ::
+      %thread
+    :~  [%channel channel.source group.source]
+        [%group group.source]
+    ==
```

Comment:
Since we're always _pre_-pending `%base`, the `%thread` case should be `~[%group %channel]` in that order, right? Otherwise the `%thread` case isn't fully ordered.

(And then, arguably, if we want to process bottom-up instead of top-down, maybe the entire order should be reversed.)